test/boost/commitlog_test.cc

```diff
@@ -494,10 +494,58 @@ SEASTAR_TEST_CASE(test_commitlog_chunk_corruption){
                         auto segments = log.get_active_segment_names();
                         BOOST_REQUIRE(!segments.empty());
                         auto seg = segments[0];
-                        return corrupt_segment(seg, rps->at(0).pos - 4, 0x451234ab).then([seg, rps] {
+                        auto cpos = rps->at(0).pos - 4;
+                        return corrupt_segment(seg, cpos, 0x451234ab).then([seg, rps, cpos] {
                             return db::commitlog::read_log_file(seg, db::commitlog::descriptor::FILENAME_PREFIX, [rps](db::commitlog::buffer_and_replay_position buf_rp) {
                                 BOOST_FAIL("Should not reach");
                                 return make_ready_future<>();
+                            }).then_wrapped([cpos](auto&& f) {
+                                try {
+                                    f.get();
+                                    BOOST_FAIL("Expected exception");
+                                } catch (commitlog::segment_truncation& e) {
+                                    // ok. We've destroyed the first chunk of the segment. This counts as a truncation
+                                    BOOST_REQUIRE(e.position() <= cpos);
+                                }
+                            });
+                        });
+                    });
+        });
+}
+
+
+SEASTAR_TEST_CASE(test_commitlog_chunk_corruption2){
+    commitlog::config cfg;
+    cfg.commitlog_segment_size_in_mb = 1;
+    return cl_test(cfg, [](commitlog& log) {
+        auto rps = make_lw_shared<std::vector<db::replay_position>>();
+        return do_until([rps]() {return rps->size() > (128*1024/8);},
```

Comment:
Yes. It is the buffer size used in fragmented temporary buffers. It is not exposed as a constant however, so ugly hard-coded.

Reply:
Are you referring to `fragmented_temporary_buffer::default_fragment_size`? I see we divide it by 8 and compare to the length of the `rps` vector, so my best guess why we compare the 2 is that each `db::replay_position` represents 8 bytes and we want to have the total length higher than the buffer size, so the fragmented tmp buffer has at least 2 buffers. If I'm mistaken please add a comment here about what we're trying to achieve. Otherwise I have a question whether the 8 bytes is in fact accurate here, after all we're writing 13 bytes ("hej bubba cow") each time we're inserting a position to `rps`?

Reply:
No. The default buffer size in commitlog.cc. 
The numbers picked ensures we write two or more chunks, given the data + overhead. Does not matter if it is exactly X entries, just we know it will not fit in one chunk. Yet are many entries.

Reply:
Ok, so the value here is exactly the buffer size used when writing chunks, but it doesn't need to be exactly that, we just know that this one is in the desired range. In that case the new comment is clear enough to me, thanks